src/app/cluster/duck/operations.ts

```diff
@@ -77,19 +80,79 @@ const addCluster = clusterValues => {
   };
 };
 
-const removeCluster = id => {
-  throw new Error('NOT IMPLEMENTED');
-  // return dispatch => {
-  //   removeClusterRequest(id).then(
-  //     response => {
-  //       dispatch(removeClusterSuccess(id));
-  //       dispatch(fetchClusters());
-  //     },
-  //     error => {
-  //       dispatch(removeClusterFailure(error));
-  //     },
-  //   );
-  // };
+const updateCluster = clusterValues => {
+  return async (dispatch, getState) => {
+    try {
+      const state = getState();
+      const { migMeta } = state;
+      const client: IClusterClient = ClientFactory.hostCluster(state);
+
+      const clusterReg = updateClusterRegistryObj(
+        clusterValues.url
+      );
+      const tokenSecret = updateTokenSecret(
+        clusterValues.token
+      );
+      const clusterRegResource = new ClusterRegistryResource(
+        ClusterRegistryResourceKind.Cluster,
+        migMeta.namespace
+      );
+      const secretResource = new CoreNamespacedResource(
+        CoreNamespacedResourceKind.Secret,
+        migMeta.configNamespace
+      );
+
+      const migClusterResource = new MigResource(MigResourceKind.MigCluster, migMeta.namespace);
+
+      const arr = await Promise.all([
+        client.patch(clusterRegResource, clusterValues.name, clusterReg),
+        client.patch(secretResource, clusterValues.name, tokenSecret),
+        client.get(migClusterResource, clusterValues.name)
+      ]);
+      const cluster = arr.reduce((accum, res) => {
+        accum[res.data.kind] = res.data;
```

Comment:
In local demo mode, this piece is failing. Would be good to add a TODO comment here to add update support for local development( if that is something that we plan on doing. not sure there is much left to develop on this so could be a waste of time. )

Reply:
Planning to work on local support with Andy.  I think local is all broken right now.